lib/OpenQA/Resource/Jobs.pm

```diff
@@ -73,9 +76,14 @@ sub job_restart {
                 next;
             }
         }
-        my $dup = $job->auto_duplicate({skip_parents => $skip_parents});
-        push @duplicated, $dup->{cluster_cloned} if $dup;
-        push @processed,  $job_id;
+        if (my $dup = $job->auto_duplicate(\%duplication_flags)) {
+            push @duplicated, $dup->{cluster_cloned};
+        }
+        else {
+            push @errors,
+              "It is not possible to restart $job_id. The job (or a dependent job) might have already a clone.";
```

Comment:
I got slightly confused by the wording. Should it be "...might have been cloned already"? Maybe we could narrow down the reason here? Since the implementation, if I read it correctly, only checks that it was done or cancelled. :thinking:

Reply:
> Since the implementation, if I read it correctly, only checks that it was done or cancelled.

That's true. But the only error which would lead to a negative return value is that single reason. (The error handling should likely still be improved to deal with exceptions. Then I would also include the actual reason.)